catchup/service.go

```diff
@@ -167,15 +174,17 @@ func (s *Service) innerFetch(fetcher Fetcher, r basics.Round) (blk *bookkeeping.
 			cf()
 		}
 	}()
-	return fetcher.FetchBlock(ctx, r)
+	peer, err := s.blocksDownloadPeerSelector.GetNextPeer()
```

Comment:
this line need to be protected with a mutex.

Reply:
There is a mutex protection inside GetNextPeer. Is that enough?

Reply:
Yes, I forgot about this one. I guess it's making your implementation even simpler.